### ISSUE TITLE
format with a single hash passed to it breaks in 1.9 and 2.0 modes

### ISSUE THREAD

**Issue by enebo** [jruby/jruby#720]
``` ruby
options = {:capture=>/\d+/}

p(" with options %p" % options)
```

We throw an argument error when we should inspect the hash and subst that into the format string.

**Comment by rtyler** [comment]
in MRI (2.1) or JRuby 9k the snippet doesn't throw an error

**Comment by kares** [comment]
marking as JRuby 1.7.x only, present on 1.7.24 : 
```
kares@sputnik:~/workspace/oss/jruby/test$ rvm use 1.7.24
Using /opt/local/rvm/gems/jruby-1.7.24
kares@sputnik:~/workspace/oss/jruby/test$ irb
jruby-1.7.24 :001 > options = {:capture=>/\d+/}
 => {:capture=>/\d+/} 
jruby-1.7.24 :002 > p(" with options %p" % options)
ArgumentError: positional args mixed with named args
	from org/jruby/RubyString.java:1228:in `%'
	from (irb):2:in `evaluate'
	from org/jruby/RubyKernel.java:1079:in `eval'
	from org/jruby/RubyKernel.java:1479:in `loop'
	from org/jruby/RubyKernel.java:1242:in `catch'
	from org/jruby/RubyKernel.java:1242:in `catch'
	from /opt/local/rvm/rubies/jruby-1.7.24/bin/irb:13:in `(root)'
jruby-1.7.24 :003 > 
jruby-1.7.24 :004 >   exit
kares@sputnik:~/workspace/oss/jruby/test$ rvm use 1.9.3
Using /opt/local/rvm/gems/ruby-1.9.3-p551
kares@sputnik:~/workspace/oss/jruby/test$ irb
1.9.3-p551 :001 > options = {:capture=>/\d+/}
 => {:capture=>/\d+/} 
1.9.3-p551 :002 > p(" with options %p" % options)
" with options {:capture=>/\\d+/}"
 => " with options {:capture=>/\\d+/}" 

```

**Comment by enebo** [comment]
This works in 9k (recently fixed as another issue in fact) and we will never fix this in 1.7.x as it is winding down....resolving.
